src/components/ListItem.jsx

```diff
@@ -1,5 +1,52 @@
+import { useEffect, useState } from 'react';
+import { updateItem, undoUpdateItem } from '../api/firebase';
+
 import './ListItem.css';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ name, listToken, itemId, dateLastPurchased }) {
+	const [wasPurchased, setWasPurchased] = useState(false);
+
+	const handleCheck = () => {
+		if (wasPurchased) {
+			undoUpdateItem(listToken, itemId);
+			setWasPurchased(false);
+		} else {
+			updateItem(listToken, itemId);
+			setWasPurchased(true);
+		}
+	};
+
+	useEffect(() => {
+		const currentDate = new Date().getTime();
+		const dayInMilliSec = 60 * 60 * 24 * 1000;
+
+		const dateLastPurchasedPlus24h = dateLastPurchased
+			? dateLastPurchased.toDate().getTime() + dayInMilliSec
+			: null;
```

Comment:
I LOVE this! I remember struggling so much with this issue and needing to use the moment.js library. You completely the trickiest part of the task so cleanly & it's easy to follow!